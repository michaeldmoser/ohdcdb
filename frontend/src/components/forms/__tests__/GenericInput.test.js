import React from 'react';
import {
    render,
    screen,
    getByText,
    getByRole,
    userEvent,
} from 'testing/library';

import { Formik } from 'formik';

import { GenericInput } from '../index';

it('should render a text input formik field', () => {
    const input = render(
        <Formik initialValues={{ title: '' }}>
            <form>
                <GenericInput name='title' label='Title' type='text' />
            </form>
        </Formik>
    );

    expect(input).toMatchSnapshot();
});

it('should render with a custom placeholder', () => {
    const input = render(
        <Formik initialValues={{ title: '' }}>
            <form>
                <GenericInput
                    name='title'
                    label='Title'
                    type='text'
                    placeholder='Enter a title'
                />
            </form>
        </Formik>
    );

    expect(input).toMatchSnapshot();
});

it('should render with a value set', () => {
    const input = render(
        <Formik initialValues={{ title: 'Testing' }}>
            <form>
                <GenericInput
                    name='title'
                    label='Title'
                    type='text'
                    placeholder='Enter a title'
                />
            </form>
        </Formik>
    );

    expect(input).toMatchSnapshot();
});

it('should set a default label', () => {
    const input = render(
        <Formik initialValues={{ title: '' }}>
            <form>
                <GenericInput name='title' type='text' />
            </form>
        </Formik>
    );

    expect(input).toMatchSnapshot();
});
