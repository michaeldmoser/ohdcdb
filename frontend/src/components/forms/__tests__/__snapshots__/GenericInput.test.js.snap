// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a text input formik field 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form>
        <div
          class="mb-3 form-floating col"
        >
          <input
            class="form-control"
            id="title"
            name="title"
            placeholder="Title..."
            type="text"
            value=""
          />
          <div
            class="invalid-tooltip"
          />
          <label
            for="title"
          >
            Title
          </label>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form>
      <div
        class="mb-3 form-floating col"
      >
        <input
          class="form-control"
          id="title"
          name="title"
          placeholder="Title..."
          type="text"
          value=""
        />
        <div
          class="invalid-tooltip"
        />
        <label
          for="title"
        >
          Title
        </label>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`should render with a custom placeholder 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form>
        <div
          class="mb-3 form-floating col"
        >
          <input
            class="form-control"
            id="title"
            name="title"
            placeholder="Enter a title"
            type="text"
            value=""
          />
          <div
            class="invalid-tooltip"
          />
          <label
            for="title"
          >
            Title
          </label>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form>
      <div
        class="mb-3 form-floating col"
      >
        <input
          class="form-control"
          id="title"
          name="title"
          placeholder="Enter a title"
          type="text"
          value=""
        />
        <div
          class="invalid-tooltip"
        />
        <label
          for="title"
        >
          Title
        </label>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`should render with a value set 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form>
        <div
          class="mb-3 form-floating col"
        >
          <input
            class="form-control"
            id="title"
            name="title"
            placeholder="Enter a title"
            type="text"
            value="Testing"
          />
          <div
            class="invalid-tooltip"
          />
          <label
            for="title"
          >
            Title
          </label>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form>
      <div
        class="mb-3 form-floating col"
      >
        <input
          class="form-control"
          id="title"
          name="title"
          placeholder="Enter a title"
          type="text"
          value="Testing"
        />
        <div
          class="invalid-tooltip"
        />
        <label
          for="title"
        >
          Title
        </label>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`should set a default label 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form>
        <div
          class="mb-3 form-floating col"
        >
          <input
            class="form-control"
            id="title"
            name="title"
            placeholder="Title..."
            type="text"
            value=""
          />
          <div
            class="invalid-tooltip"
          />
          <label
            for="title"
          >
            Title
          </label>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form>
      <div
        class="mb-3 form-floating col"
      >
        <input
          class="form-control"
          id="title"
          name="title"
          placeholder="Title..."
          type="text"
          value=""
        />
        <div
          class="invalid-tooltip"
        />
        <label
          for="title"
        >
          Title
        </label>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
